File: common/src/main/java/com/matejdro/wearremotelist/receiverside/conn/WatchSingleConnection.java
package com.matejdro.wearremotelist.receiverside.conn;

import com.google.android.gms.common.api.GoogleApiClient;
import com.google.android.gms.wearable.Node;
import com.google.android.gms.wearable.Wearable;

import java.util.Collections;
import java.util.Comparator;
import java.util.List;

public class WatchSingleConnection extends PlayServicesConnectionToProvider
{
    /**
     * {@inheritDoc}
     */
    public WatchSingleConnection(GoogleApiClient googleApiClient)
    {
        super(googleApiClient);
    }

    @Override
    public String getProviderNodeId()
    {
        List<Node> connectedNodes = Wearable.NodeApi.getConnectedNodes(getGoogleApiClient()).await().getNodes();
        if (connectedNodes == null || connectedNodes.isEmpty())
            return null;

        Collections.sort(connectedNodes, NodeNearbyComparator.INSTANCE);
        return connectedNodes.get(0).getId();
    }

    private static class NodeNearbyComparator implements Comparator<Node>
    {
        public static final NodeNearbyComparator INSTANCE = new NodeNearbyComparator();

        @Override
        public int compare(Node a, Node b)
        {
            int nearbyA = a.isNearby() ? 1 : 0;
            int nearbyB = b.isNearby() ? 1 : 0;
            return nearbyB - nearbyA;
        }
    }
}

File: common/src/main/java/com/matejdro/wearremotelist/receiverside/conn/WatchSingleConnection.kt
package com.matejdro.wearremotelist.receiverside.conn

import com.google.android.gms.common.api.GoogleApiClient
import com.google.android.gms.wearable.Node
import com.google.android.gms.wearable.Wearable
import java.util.Collections

class WatchSingleConnection
/**
 * {@inheritDoc}
 */
    (googleApiClient: GoogleApiClient?) : PlayServicesConnectionToProvider(googleApiClient) {

    override fun getProviderNodeId(): String? {
        val connectedNodes = Wearable.NodeApi.getConnectedNodes(googleApiClient).await().nodes
        if (connectedNodes.isEmpty()) return null
        Collections.sort(connectedNodes, NodeNearbyComparator.INSTANCE)
        return connectedNodes[0].id
    }

    private class NodeNearbyComparator : Comparator<Node> {
        override fun compare(a: Node, b: Node): Int {
            val nearbyA = if (a.isNearby) 1 else 0
            val nearbyB = if (b.isNearby) 1 else 0
            return nearbyB - nearbyA
        }

        companion object {
            val INSTANCE = NodeNearbyComparator()
        }
    }
}
